### DISCUSSION TITLE
suggestion: add config flag to deno lint and deno fmt

### DISCUSSION THREAD

**Discussion by DjDeveloperr** [denoland/deno#8778] [Ideas]
EDIT: It was added in v1.14 🎉 

Hi!

I was reading through #dev-lint (Discord Channel) conversation. This is about adding config flag to `deno lint` and `deno fmt`.

Honestly having a config isn't bad, actually our deno project still depends on Node. Why? we're using eslint because deno lint cannot be configured the way we want. We have to use prettier because deno fmt cannot be configured and the default style is not what we want it to be.
In my opinion, it would have been way straightforward if `deno lint` and `deno fmt` supported config - no need to use external tooling, deno ships with it.

I read through https://github.com/denoland/deno/issues/3827 too, nothing reasonable I could find.

I'm not able to understand the reason for not adding config. It's just optional, use it if you want, not otherwise..

P.S: Its just a suggestion, I don't want to bring up any argument or something.

**Comment by lucacasonato** [comment]
What exactly do you want to configure?

**Reply by DjDeveloperr** [reply]
For formatter, we want to configure semi colons, tab-width, column-width, and quotes.
And for linter, just want to configure the set of rules (disable some entirely).

**Reply by lucacasonato** [reply]
Can you be more specific about what rules you want to disable for the linter?

**Reply by DjDeveloperr** [reply]
camelCase, require-await, no-inferable-types mostly.

**Reply by Soremwar** [reply]
@DjDeveloperr Not using await in an async function is kinda pointless to be fair. And why would you want to have redundant types in your code anyway? Doesn't that increase the length of your code pointlessly?

As for removing `camelCase` +1 for that. No code should be flagged as incorrect just cause it uses one _pretty_ name over another

PD: Not trying to spark any discussion, just arguing that (excluding `camelCase`) deno-lint has really good defaults.

**Reply by DjDeveloperr** [reply]
> @DjDeveloperr Not using await in an async function is kinda pointless to be fair. And why would you want to have redundant types in your code anyway? Doesn't that increase the length of your code pointlessly?
> 
> As for removing `camelCase` +1 for that. No code should be flagged as incorrect just cause it uses one _pretty_ name over another
> 
> PD: Not trying to spark any discussion, just arguing that (excluding `camelCase`) deno-lint has really good defaults.

You’re right. `camelCase` is mostly bothering me. 

I forgot to mention about plugins last time, I had to go. What about being able to define custom set of rules using something like plugins? Like in eslint.

**Reply by jtoppine** [reply]
> @DjDeveloperr Not using await in an async function is kinda pointless to be fair. And why would you want to have redundant types in your code anyway? Doesn't that increase the length of your code pointlessly?

In class definitions, I personally consider the async keyword in class methods part of the API design and documentation, or self-documenting code. It immediately tells that the method is _meant_ to be called in async manner, even if the _current implementation_  doesn't. 

Say, you have a Cache base class that implements simple non-async in-memory cache, or perhaps has just dummy "no-op" placeholder methods. The cache API is nevertheless designed to be async, so that it can later be extended with async LocalFileCache, S3Cache or whatever specialized truly async implementations. I think it is desireable to have the Cache base class methods also rock the async keyword, and pretty annoying that deno lsp forces one to litter the code with // deno-lint-ignore require-await

**Comment by bartlomieju** [comment]
This is resolved as of Deno v1.14 https://deno.com/blog/v1.14#customization-options-for-deno-lint-and-deno-fmt
